code/modules/mob/living/carbon/monkey/diona.dm

```diff
@@ -135,6 +136,17 @@
 				return 1
 	return ..()
 
+/mob/proc/sample_can_reach(mob/M as mob, sting_range = 1)
+	if(M.loc == src.loc)
+		return 1 //target and source are in the same thing
+	if(!isturf(src.loc) || !isturf(M.loc))
```

Comment:
This check will prevent a Diona that's being carried from taking samples. That seems unintended, though I don't know current behavior offhand.